### DISCUSSION TITLE
Installation on Linux: How to do it in Ansible (package manager) ?

### DISCUSSION THREAD

**Discussion by interfan7** [go-graphite/carbonapi#804] [Q&A] [answered]
We use Ansible to install Linux packages such as RPM and DEB. 
Ansible supports such package managers. 

Reading the installation instructions here I've seen only a method of [downloading a script from curl and executing it](https://packagecloud.io/go-graphite/stable/install). 

There are no built packages in assets [unlike in go-carbon](https://github.com/go-graphite/go-carbon/releases/tag/v0.17.1). 

I'm not familiar with _packagecloud_ and not sure whether it provides a "clean" methodology to manage package/installations on a Linux machine.

**Comment by Civil** [comment]
that script basically do:
1. Downloading corresponding GPG key
2. Generate sources file for package manager

You can manually obtain GPG key from the URL: https://packagecloud.io/go-graphite/stable/gpgkey

And you can generate config for your distro by querying URL: 
```
https://packagecloud.io/install/repositories/go-graphite/stable/config_file.list?os=${os}&dist=${dist}&source=script
```

Just replace os and dist with your OS and distro version.

Packagecloud also have ansible role available: https://blog.packagecloud.io/introducing-the-packagecloud-ansible-role/

If it is better to attach packages to the release, I can modify CI scripts to do that for next release (I don't think I'll retrospectively do that though).

**Comment by tbauriedel** [comment, marked as answer]
Since graphite-api has not been maintained for years and the installation of graphite-web is not always so nice, I have now looked at go-carbon and carbonapi as an alternative.

I am a person who likes to write things in Ansible so that I can reuse them. In my opinion, the learning effect is also much higher, as you have to deal intensively with the tool in order to write it in Ansible.

Long story short:
The evaluation of the two tools not only gave me a good and uncomplicated alternative for graphite-web and carbon, but also led to a small Ansible collection. It is not yet fully developed and still needs some love. But for the moment and for my use case it works for now ;) .

I have published it on github so that everyone can use it.
https://github.com/tbauriedel/ansible-collection-gographite

I hope I can make life a little bit easier for you and other readers with this collection!
I am of course very grateful for any feedback.
